### PR TITLE
Add type-hints for _vformat in string.Formatter

### PR DESCRIPTION
**Pull request by ischaojie**
Add type-hints for `_vformat` method in `string.Formatter`.

When add type-hints for
https://github.com/pypa/hatch/blob/b0911bb0eaa8d331c24eda940b97bf244ecd5ac3/backend/src/hatchling/utils/context.py#LL119 part,  mypy report `Cannot access member "_vformat" for type` error.

But `_vformat` not appear in API document, I'm not sure should add this.

### CODE REVIEW

File: stdlib/string.pyi [hunk 1]
```diff
@@ -64,6 +64,15 @@ class Formatter:
...
     def vformat(self, format_string: str, args: Sequence[Any], kwargs: Mapping[str, Any]) -> str: ...
+    def _vformat(
+        self,
+        format_string: str,
+        args: Sequence[Any],
+        kwargs: Mapping[str, Any],
+        used_args: Sequence[int | str],
+        recursion_depth: int,
+        auto_arg_index: int = ...,
+    ) -> str | bool: ...
```
Comment (JelleZijlstra) [inline stdlib/string.pyi:75]:
```suggestion
    ) -> str: ...
```
Don't see anything in the code that returns a bool.

File: stdlib/string.pyi [hunk 2]
```diff
@@ -64,6 +64,15 @@ class Formatter:
     ) -> LiteralString: ...
     @overload
     def vformat(self, format_string: str, args: Sequence[Any], kwargs: Mapping[str, Any]) -> str: ...
+    def _vformat(
+        self,
+        format_string: str,
+        args: Sequence[Any],
+        kwargs: Mapping[str, Any],
+        used_args: Sequence[int | str],
```
Comment (JelleZijlstra) [inline stdlib/string.pyi:72]:
```suggestion
        used_args: set[int | str],
```
It does `used_args.add`.

File: stdlib/string.pyi [hunk 3]
```diff
@@ -64,6 +64,15 @@ class Formatter:
     ) -> LiteralString: ...
     @overload
     def vformat(self, format_string: str, args: Sequence[Any], kwargs: Mapping[str, Any]) -> str: ...
+    def _vformat(
```
Comment (JelleZijlstra) [inline stdlib/string.pyi:67]:
```suggestion
    def _vformat(  # undocumented
```
Following our convention.

File: stdlib/string.pyi (hunk 2 above)
Comment (ischaojie) [reply stdlib/string.pyi:72]: I notice `self.check_unused_args(used_args, args, kwargs)` in method `vformat`, but :
```pyi
def check_unused_args(self, used_args: Sequence[int | str], args: Sequence[Any], kwargs: Mapping[str, Any]) -> None: ...
```
Maybe need add `@overload` for `check_unused_args` stub?

File: stdlib/string.pyi (hunk 1 above)
Comment (ischaojie) [reply stdlib/string.pyi:75]: Thanks, I misread.

File: stdlib/string.pyi (hunk 2 above)
Comment (JelleZijlstra) [reply stdlib/string.pyi:72]: I think that annotation is just wrong. It should also be `set[int | str]`.

File: stdlib/string.pyi (hunk 3 above)
Comment (ischaojie) [reply stdlib/string.pyi]: resolved.

File: stdlib/string.pyi (hunk 2 above)
Comment (ischaojie) [reply stdlib/string.pyi]: resolved.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**